### ISSUE TITLE
Typo in multi-line log example

### ISSUE THREAD

**Issue by pranavmarla** [kubernetes/community#5984]
In the _Log Format_ subsection of the [CRI log format doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/kubelet-cri-logging.md#proposed-solution), there is an example of how a multi-line log looks. I believe there is a typo in that example -- specifically, the timestamps appear to be off.

Here is the example, representing two logs, where (single-line) log 1 is the first line, and (multi-line) log 2 is split across the last 3 lines, with a blank line separating the two logs for clarity:
```
2016-10-06T00:17:09.669794202Z stdout F The content of the log entry 1

2016-10-06T00:17:09.669794202Z stdout P First line of log entry 2
2016-10-06T00:17:09.669794202Z stdout P Second line of the log entry 2
2016-10-06T00:17:10.113242941Z stderr F Last line of the log entry 2
```

As far as I'm aware, each line of a multi-line log entry should be assigned the same timestamp, which means that the last 3 lines (log 2) should share the same timestamp, and the first line (log 1) should have a different timestamp.
Instead, log 1 shares the same timestamp with the first two lines of log 2 , and the last line of log 2 has a different timestamp.

**Comment by SARANYA233** [comment]
I am new to this open-source contribution, so I want to work on this issue. Can I work on this issue?

**Comment by lakshya8066** [comment]
Hey @SARANYA233 you can self assign yourself by commenting "/assign" to this thread :)

**Comment by SARANYA233** [comment]
> Hey @SARANYA233 you can self assign yourself by commenting "/assign" to this thread :)

/assign

**Comment by SARANYA233** [comment]
@pranavmarla 

> TIMESTAMP STREAM TAG CONTENT

2016-10-06T00:17:09.669794202Z stdout P:TAG1:TAG2 log content 1 
2016-10-06T00:17:09.669794203Z stdout P:TAG1:TAG2 log content 2

Is this what you were saying about?
Actually, I m new to this concept so somehow I tried to collect the concept 
If it is wrong plz explain to me in detail what the content is about.

**Comment by pranavmarla** [comment]
@SARANYA233 
> TIMESTAMP STREAM TAG CONTENT
2016-10-06T00:17:09.669794202Z stdout P:TAG1:TAG2 log content 1
2016-10-06T00:17:09.669794203Z stdout P:TAG1:TAG2 log content 2
Is this what you were saying about?

Sorry, not sure I understand.  Did you copy that from somewhere? When you say "Is _this_ what you were saying about?", what are you referring to?

> Actually, I m new to this concept so somehow I tried to collect the concept
If it is wrong plz explain to me in detail what the content is about.

I explained the issue in the first post up top -- can you tell me which part you didn't understand? Then I can try and explain that part again.

**Comment by SARANYA233** [comment]
@pranavmarla Actually, I understood that you were telling there is an issue in *log entry 2* rite. so when I was studying on this issue I found this,

2016-10-06T00:17:09.669794202Z stdout P:TAG1:TAG2 log content 1
2016-10-06T00:17:09.66979420**3**Z stdout P:TAG1:TAG2 log content 2

In log content 2, 2 -> 3, I have bolded it.
whether it is right or should I do any execution work.

**Comment by pranavmarla** [comment]
@SARANYA233 Hmm, I THINK that's right?

Let me try to explain again: Basically, if you go [here](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/kubelet-cri-logging.md#proposed-solution), you will see this section:

> 2016-10-06T00:17:09.669794202Z stdout F The content of the log entry 1
2016-10-06T00:17:09.669794202Z stdout P First line of log entry 2
2016-10-06T00:17:09.669794202Z stdout P Second line of the log entry 2
2016-10-06T00:17:10.113242941Z stderr F Last line of the log entry 2

Currently, the first 3 lines have the same timestamp, and the last line has a different timestamp.
Instead, I think the first line should have a different timestamp, and the last 3 lines should have the same timestamp.

It does not matter what the timestamps are but, for clarity, it would probably be better if the two timestamps do not look similar (so that it is easy to see that they are different).
Eg.
`00:17:09.669794202` looks very similar to `00:17:09.669794203`, so it is hard to tell that they are different -- so, choosing these two timestamps would not be a good choice.
By contrast, it is easier to see that `00:17:09.669794202` and `00:17:10.113242941` are different -- so, choosing something like these two would be a good choice.

Hope that helps!

**Comment by SARANYA233** [comment]
@pranavmarla yeah I understood the final log might be like this rite

2016-10-06T00:17:09.669794202Z stdout F The content of the log entry 1
2016-10-06T00:17:09.113242941Z stdout P First line of log entry 2
2016-10-06T00:17:09.113242941Z stdout P Second line of the log entry 2
2016-10-06T00:17:10.113242941Z stderr F Last line of the log entry 2

How we will know whether what we did is right or wrong?

**Comment by pranavmarla** [comment]
Hey @SARANYA233

> yeah I understood the final log might be like this rite
>
>2016-10-06T00:17:09.669794202Z stdout F The content of the log entry 1
2016-10-06T00:17:09.113242941Z stdout P First line of log entry 2
2016-10-06T00:17:09.113242941Z stdout P Second line of the log entry 2
2016-10-06T00:17:10.113242941Z stderr F Last line of the log entry 2

No, the last line of log 2 still has a different timestamp from the two lines above it -- I'm saying that I think it should look something like this:

2016-10-06T00:17:09.669794202Z stdout F The content of the log entry 1
**2016-10-06T00:17:10.113242941Z** stdout P First line of log entry 2
**2016-10-06T00:17:10.113242941Z** stdout P Second line of the log entry 2
**2016-10-06T00:17:10.113242941Z** stderr F Last line of the log entry 2

> How we will know whether what we did is right or wrong?
That is a good question lol. I'm not an expert on the CRI log format, so I cannot guarantee that my theory (that every line of a multi-line log should have the identical timestamp) is correct -- maybe one of the maintainers, or someone with more knowledge on this in general, can confirm this?
If you end up opening a pull request, I assume they will check this anyway, as part of reviewing your pull request.

**Comment by mrbobbytables** [comment]
I commented on the PR, but as an FYI the documentation in the design-proposal directory is considered to be in an [archived state  and no longer accepting updates](https://github.com/kubernetes/community/tree/master/contributors/design-proposals#kubernetes-design-documents-and-proposals) (PRs are blocked from being merged via the `do-not-merge/blocked-paths` label). Any future changes should be made either via KEPs or in the SIG specific documentation under the [devel docs](https://github.com/kubernetes/community/tree/master/contributors/devel).

**Comment by pranavmarla** [comment]
Thanks for adding that here as well @mrbobbytables !

Incidentally, my assumption behind this issue (and @SARANYA233 's later pull request) was that, when a multi-line log is split into multiple log entries by a CRI-implementing container runtime (eg. containerd), each log entry (i.e. each line) is assigned the same timestamp.

However, I recently had the chance to examine some log entries produced by containerd and, contrary to my assumption, every log entry / line had a different timestamp, even though they came from the same log!

So, although the [CRI log format doc](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/node/kubelet-cri-logging.md#proposed-solution) could definitely be clearer about this, it doesn't appear to actually be wrong as I initially thought -- so, I'll be closing this issue as well.
